### PR TITLE
(PA-183, RE-5735) PFTW logic migration to PA

### PR DESCRIPTION
**Pull request by melissa**

### CODE REVIEW

**Comment by melissa** [comment]
Requires https://github.com/puppetlabs/vanagon/pull/278

File: configs/components/nssm.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+component "nssm" do |pkg, settings, platform|
+  pkg.load_from_json("configs/components/nssm.json")
+
+  # Because we're unpacking a zip archive, we need to set the path to the executable.
+  # We don't automatically have this set on windows, unfortunately. We need to set the
+  # path to have access to 7za.
+  pkg.environment "PATH" => "/cygdrive/c/ProgramData/chocolatey/tools:$$PATH" if platform.is_windows?
```
Comment (shermdog) [inline configs/components/nssm.rb]: Is this something we should be setting at the platform level?  Seems like we may need this in other components.

Comment (melissa) [inline configs/components/nssm.rb]: Yeah, probably. Do we have the ability to set it at the platform level?

Comment (shermdog) [inline configs/components/nssm.rb]: Doesn't look like we can :(  Obviously that negate my comment.  Let's discuss offline if we should add a ticket to vanagon for that in the future.

**Comment by shermdog** [comment]
:+1:

File: configs/components/nssm.rb (hunk 1 above)
Comment (MikaelSmith) [inline configs/components/nssm.rb]: You could have provisioning cat something in the ssh profile.

**Comment by MikaelSmith** [comment]
This seems ok. Is Windows going to have an implementation of `install_service`?

**Comment by melissa** [comment]
@MikaelSmith see https://github.com/puppetlabs/puppet-agent/pull/519 for `install_service`

**Comment by melissa** [comment]
@MikaelSmith also https://github.com/puppetlabs/vanagon/blob/ab4841e87c0d58f4762208455f8f1047a8c45d7c/lib/vanagon/component/dsl.rb#L187-L188 for `install_service` on windows

File: configs/components/nssm.rb (hunk 1 above)
Comment (melissa) [inline configs/components/nssm.rb]: I think there's a ticket somewhere for setting up a base environment... I can't find it at the moment, but this is definitely something that I'd like to see happen at some point

Comment (stahnma) [inline configs/components/nssm.rb]: The only we  to do this globally currently that I know of is have a file with the stuff you want that many/all of the components load in and eval. Not ideal. So, IOW, LGTM right now.

**Pull request merged**